ceph_iscsi_config/gateway.py

```diff
@@ -63,7 +63,7 @@ def __init__(self, logger, iqn, gateway_ip_list, enable_portal=True):
             # if the ip list provided doesn't match any ip of this host, abort
             # the assumption here is that we'll only have one matching ip in
             # the list!
-            matching_ip = set(gateway_ip_list).intersection(ipv4_addresses())
+            matching_ip = set(gateway_ip_list).intersection(ip_addresses())
```

Comment:
Is gateway_ip_list supposed to have the ipv6 addresses with [] or without? As far as I can tell it looks like normally it is coming from the config object, but in ceph-ansible it comes from the user.

I ask because I think there might be a mismatch. 

Here ip_addresses returns ipv6 addrs without []s so for the intersection to match, gateway_ip_list has to have addrs without []s.

```
            matching_ip = set(gateway_ip_list).intersection(ip_addresses())
...
            self.active_portal_ip = list(matching_ip)[0]
...
           gateway_metadata = {"portal_ip_address": self.active_portal_ip,
```

We then store the addr for the "portal_ip_address" key without []s.


But then below on line 359 we do:
```

        if canonicalize_ip(config.config["gateways"][owning_gw]["portal_ip_address"]) == \
                tpg_ip_address:

```
and we have done canonicalize_ip on tpg_ip_address, so that check is expecting []s.

Reply:
> normally it is coming from the config object, but in ceph-ansible it comes from the user.

For the gwcli case the ip address come directly from the user too right? Should we add a check in GWTarget __init__ to make sure the passed in addrs are the format we require or just fix it up for the user?

Reply:
I’d prefer to drop the square brackets since we aren’t including a port so there is no need to disambiguate. The goal for “tgt_ip_address” was that it would have brackets if it came from rtslib (since that’s how it is) or just added the brackets if we pulled it from the config and compare it to a bracketed version.

Reply:
Ok to me.

Reply:
@mikechristie OK -- I normalize to IP addresses on input (and extraction from rtslib) and normalize to IP literals when sending values to rtslib.